#161255377 Users should be able to like a specific comment

AnguleMathias opened the pull request:
#### What does this PR do?
* Users should be able to like a specific comment

#### Description of tasks completed
* Tests for the like comments.
* Added the comment_like field in the database.
* Added the like count in the CommentSerializer.

#### Working endpoint 
            i.  `api/articles/<slug>/comment/<pk>/like/`
           

#### How can this PR be manually tested?
* clone the branch.
* To like a comment: `PUT api/articles/<slug>/comment/<pk>/like/` 
**NOTE:** Repeating the above step will unlike the comment.
* run tests using python manage.py test. All tests should pass.

#### What are the relevant pivotal tracker stories?
* #161255377 - [Users should be able to like a specific comment](https://www.pivotaltracker.com/story/show/161255377)

#### Screenshots
###### 1. Like a comment
<img width="1361" alt="screenshot 2018-11-14 at 20 29 59" src="https://user-images.githubusercontent.com/39617195/48500585-41914c80-e84c-11e8-87fe-863520f23674.png">

###### 2. Unlike a comment
<img width="1362" alt="screenshot 2018-11-14 at 20 29 49" src="https://user-images.githubusercontent.com/39617195/48500602-481fc400-e84c-11e8-8234-8d921abe6ed1.png">

AnguleMathias committed:
Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models
- Added comment-likes and comment-dislikes method to comment serializer class
- Automatic updates to  migrations
- Deleted .DS_Store

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models
- Added comment-likes and comment-dislikes method to comment serializer class
- Automatic updates to  migrations
- Deleted .DS_Store

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models tests
- Update comment-likes and comment-dislikes method to comment serializer class
- Updates to migrations

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like/dislike comments to articles

- Added comment-likes and comment_dislikes fields to models tests
- Update comment-likes and comment-dislikes method to comment serializer class
- Updates to migrations

[Feature #161255377]

Jwesh reviewed (approved):
some of your lines are more than 79 characters... check on that..

maxwellgithinji reviewed (approved):
LGTM

3N61N33R commented on authors/apps/articles/tests/test_like_or_dislike_comments.py:
```diff
@@ -0,0 +1,208 @@
+            "time_to_read": 1,
+            "tags": ["TDD"]
+        }
+
```
since you're importing from .base you can use the article data defined there instead of having to define another set of data

AnguleMathias replied:
Check for changes

Commit:
#161255376 comment track edit history (#32)

* Feature(track comment history):Comments should be able to track edit history

- added CommentHistory model
- modified comment update to store history

[Feature #161255376]

* Feature(track comment history):Comments should be able to track edit history

    - remove debug statements
    - add docstrings

    [Feature #161255376]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Feedback implementation

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Feedback implementation

[Feature #161255377]

3N61N33R committed:
#161255369 Add Search Functionality and Custom filtering based on parameters (#28)

* feature(search_filter): search, filter functionality

-add app search and filter functionality

[feature #161255369]

* feature(search_filter): search, filter functionality

-fix list articles bug
-add doctstrings

[feature #161255369]

* feature(filter_search): filter, search functionality

-add docstrings
-remove repeated classes

[feature #161255369]

njeri-ngigi committed:
#161255374 Highlight and Comment on any text in an article (#31)

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(Highlight and comment on text): Add CRUD for highlights and comments on text
- Add CRUD functions for higlighting and commenting on any text of an article
- Add tests for these CRUD functions
[Feature #161255374]

* Feature(#161255374 Highlighting and commenting on text): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255374]

njeri-ngigi committed:
#161255370 Users should be able to see the time it takes to read an article (#34)

* Feature(#161255370 Time to read): Calculate time to read
- Add method to calculate time to read an article
- Add test for calculating time to read an article
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

* Feature(#161255370 Time to read): Lint code
- Adjust code to match pep8 linting standards
[Feature #161255370]

3N61N33R committed:
#161255368 Pagination support for articles (#35)

* feature(pagination): pagination for articles

-enable pagination for articles

[feature #161255368]

* feature(pagination): article pagination functionality

-override pagination response
-fix search and filter functionality
- resolve merge conflicts

[feature #161255368]

* feature(pagination): article pagination functionality

-remove import error

[feature #161255368]

* feature(pagination): article pagination functionality

-fix tests

[feature #161255368]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Rebase commit

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Rebase commit

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Rebase commit

[Feature #161255377]

AnguleMathias committed:
Feature(Comment likes):Users should be able to like comments to articles

- Refactor tests

[Feature #161255377]
